### PR TITLE
support proposed new FoT-capable methods

### LINKED ISSUE #835: Support of fee-on-transfer tokens
**Is your feature request related to a problem? Please describe.**
The Uniswap v2 front-end doesn't support deflationary tokens. (tokens that burn a % of each transaction, like the one I'm contributing to, "BSOV" (https://bsov.io).
BSOV Token Request: https://github.com/Uniswap/uniswap-frontend/issues/836
Deflation is very important for the deflationist community, and many of us wish that BSOV keeps on burning, while trading on-chain.

**Describe the solution you'd like**
Possibly like @NoahZinsmeister described to me in the Discord: A feature in the Uniswap v2 front-end that lets the user "set their slippage to whatever % deflation they expect from their token".

**Describe alternatives you've considered**
When there is a token request of a deflationary token on Github, then there could be a custom parameter that could be prefixed for easier implementation of deflationary tokens in the Uniswap v2 front-end, so the user would not need to set their slippage for **every** trade, to match the % deflation.

**Additional context**
While I don't know how to develop these things, then there are several things I can help you with, like paying you some BSOV or doing tasks revolving Media & Communication (like design and videos).

### PR DESCRIPTION
**Pull request by NoahZinsmeister**
closes #835

### CODE REVIEW

File: src/hooks/useSwapCallback.ts [hunk 1]
```diff
@@ -90,112 +90,148 @@ export function useSwapCallback(
...
-          }
+        )
+      )
 
-          return response.hash
-        })
-        .catch(error => {
-          console.error(`Swap or gas estimate failed`, error)
-          throw error
+      const indexOfSuccessfulEstimation = safeGasEstimates.findIndex(safeGasEstimate =>
```
Comment (moodysalem) [inline src/hooks/useSwapCallback.ts:160]: https://caniuse.com/#search=findindex

File: src/hooks/useSwapCallback.ts [hunk 2]
```diff
@@ -90,112 +90,148 @@ export function useSwapCallback(
...
-                slippageAdjustedInput.toSignificant(3) +
-                ' ' +
-                trade.inputAmount.token.symbol +
-                ' for ' +
-                slippageAdjustedOutput.toSignificant(3) +
-                ' ' +
-                trade.outputAmount.token.symbol
+      const safeGasEstimates = await Promise.all(
+        methodNames.map(methodName =>
+          routerContract.estimateGas[methodName](...args, value ? { value } : {})
```
Comment (moodysalem) [inline src/hooks/useSwapCallback.ts:152]: a hook for `useGasEstimate` would be generally useful, but would require constructing all these args on every render, which is probably ok for the improved UX (we can disable the button and show an error if we can't get a gas estimate from either swap)

File: src/hooks/useSwapCallback.ts (hunk 1 above)
Comment (NoahZinsmeister) [reply src/hooks/useSwapCallback.ts:160]: oof, will change this

File: src/hooks/useSwapCallback.ts (hunk 2 above)
Comment (NoahZinsmeister) [reply src/hooks/useSwapCallback.ts:152]: yeah i like it, i'll start to refactor

File: src/hooks/useSwapCallback.ts (hunk 1 above)
Comment (NoahZinsmeister) [reply src/hooks/useSwapCallback.ts:160]: wait you linked TypedArray, Array is fine: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex

we're using findIndex in 2 other places in the codebase as well

Comment (moodysalem) [reply src/hooks/useSwapCallback.ts:160]: thought i'd seen it elsewhere, sgtm.

File: src/hooks/useSwapCallback.ts (hunk 2 above)
Comment (NoahZinsmeister) [reply src/hooks/useSwapCallback.ts:152]: going to track this in #872

**Pull request merged**